agent/provider/src/payments/payments.rs

```diff
@@ -389,49 +391,54 @@ impl Handler<ActivityCreated> for Payments {
     type Result = anyhow::Result<()>;
 
     fn handle(&mut self, msg: ActivityCreated, ctx: &mut Context<Self>) -> Self::Result {
-        if let Some(agreement) = self.agreements.get_mut(&msg.agreement_id) {
-            log::info!(
-                "Payments - activity [{}] created. Start computing costs.",
-                &msg.activity_id
-            );
+        let agreement = self
+            .agreements
+            .get_mut(&msg.agreement_id)
+            .ok_or(anyhow!(
+                "Agreement [{}] wasn't registered.",
+                &msg.agreement_id
+            ))
+            .log_warn()?;
 
-            // Sending UpdateCost with last_debit_note: None will start new
-            // DebitNotes chain for this activity.
-            let msg = UpdateCost {
-                invoice_info: DebitNoteInfo {
-                    agreement_id: msg.agreement_id.clone(),
-                    activity_id: msg.activity_id.clone(),
-                    payment_deadline: None, // Will be added in UpdateCost handler.
-                },
-            };
+        log::info!(
+            "Payments - activity [{}] created. Start computing costs.",
+            &msg.activity_id
+        );
 
-            // Add activity to list and send debit note after update_interval.
-            agreement.add_created_activity(&msg.invoice_info.activity_id);
-            ctx.notify_later(msg, agreement.update_interval);
+        // Sending UpdateCost with last_debit_note: None will start new
+        // DebitNotes chain for this activity.
+        let msg = UpdateCost {
+            invoice_info: DebitNoteInfo {
+                agreement_id: msg.agreement_id.clone(),
+                activity_id: msg.activity_id.clone(),
+                payment_deadline: None, // Will be added in UpdateCost handler.
```

Comment:
Why not fill it here?

Reply:
`UpdateCost` is called later. Check `ctx.notify_later` a few lines below.